doc/users_guide.rst

```diff
@@ -0,0 +1,368 @@
+You should now have all of the required dependencies installed. From now on, you
+can use the LISA shell, which provides a convenient set of commands for easy
+access to many LISA related functions (done automatically by Vagrant)::
+
+  source init_env
+
+Run ``lisa-help`` to see an overview of the provided LISA commands.
+
+Notebooks
+=========
+
+The LISA shell can simplify starting an Jupyter notebook server::
+
+  [LISAShell lisa] \> lisa-jupyter start
+
+  Starting Jupyter Notebooks...
+
+  Notebook server configuration:
+    URL        :  http://127.0.0.1:8888/?token=b34F8D0e457BDa570C4A6D7AF113CB45d9CcAF44Aa7Cf400
+    Folder     :  /data/work/lisa/ipynb
+    Logfile    :  /data/work/lisa/ipynb/server.log
+    PYTHONPATH :
+	  /data/work/lisa/modules_root/
+
+
+  Notebook server task: [4] 30177
+
+Note that the ``lisa-jupyter`` command allows you to specify interface and
+port in case you have several network interfaces on your host::
+
+  lisa-jupyter start [interface [port]]
+
+The URL of the main folder served by the server is printed on the screen.
+By default it is http://127.0.0.1:8888/. Once the server is started you can
+have a look at the provided tutorial notebooks are accessible by following this
+`link <http://127.0.0.1:8888/notebooks/examples/typical_experiment.ipynb>`__.
+This initial tutorial can be seen (but not executed) also on `github
+<https://github.com/ARM-software/lisa/blob/next/ipynb/examples/typical_experiment.ipynb>`__.
+
+
+Transitioning from LISA legacy
+==============================
+
+A big refactoring effort was started in mid 2018, which produced a lot of
+(much needed) changes. If you are used to using LISA before the refactoring came
+into place, this guide is for you.
+
+Global changes
+++++++++++++++
+
+Project structure
+-----------------
+
+* ``$repo/libs/utils`` is now ``$repo/lisa/``. ``$repo/libs/wlgen`` has also been
+  moved to that location.
+* :mod:`devlib`, :mod:`trappy`, :mod:`bart` and :mod:`wa` are
+  now under ``$repo/external/``. Git subtrees are now used instead of submodules.
+* All non self-tests have been moved from ``$repo/tests`` to ``$repo/lisa/tests``
+
+Python 3
+--------
+
+With Python 2 end of life drawing near, we decided to bridge the gap and move over
+to Python 3. Unlike :mod:`devlib` or :mod:`trappy`, we didn't go for compatibility
+with both Python 2 and Python 3 - LISA is now Python 3 only.
+
+Imports
+-------
+
+LISA legacy used implicit relative imports with a bit of dark magic to hold
+everything together. Say you want to import the :class:`lisa.trace.Trace` class
+found in ``lisa/trace.py``, previously you would do it like so::
+
+  from trace import Trace
+
+However, implicit relative imports are dangerous - did you know :class:`trace.Trace`
+exists in Python's standard library? This means that with the previous setup, the LISA
+module would shadow the standard library's. The above import done in a non-LISA
+environment would import something completely different!
```

Comment:
s/would import/would have imported/

That only applied to lisa-legacy, Python 3 does not have implicit relative import anymore so in lisa-next, it would always import the stdlib one.